framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -3442,4 +3547,32 @@ public ProcessingEnvironment getProcessingEnv() {
     public CFContext getContext() {
         return checker;
     }
+
+    /**
+     * Factory method to reflectively set {@link AnnotatedTypeFactory#viewpointAdaptor}. By default,
+     * this method gets the name of type system from the canonical name of "this", and concatenate
+     * it with "ViewpointAdaptor". If the concrete type system has a different naming mechanism,
+     * then developer of that type system should override this method to instantiate it correctly.
+     * If the target class can be instantiated successfully, it's used. Otherwise, internal error is
+     * triggered and will be error aborted.
+     *
```

Comment:
Already documented. Passing `checker.withViewpointAdaptation()` as a parameter makes the developer's life easier. Since reflectively creating subclass of ViewpointAdaptor is in the AnnotatedTypeFactory, we do need this flag to decide if we instantiate ViewpointAdaptor or not. If we don't do reflextion mechanism, developer would need to override a method to only return the their own ViewpointAdadptor, which doesn't look nicer than just overriding `checker.withViewpointAdaptation()` to return true, and let CF automatically pick up ViewpointAdaptor subclass.